src/mesido/workflows/io/write_output.py

```diff
@@ -954,19 +954,26 @@ def _name_to_asset(name):
                 *self.energy_system_components.get("heat_source", []),
                 *self.energy_system_components.get("ates", []),
                 *self.energy_system_components.get("heat_buffer", []),
+                *self.energy_system_components.get("heat_pump", []),
             ]:
                 asset = _name_to_asset(name)
                 asset_placement_var = self._asset_aggregation_count_var_map[name]
                 placed = np.round(results[asset_placement_var][0]) >= 1.0
                 max_size = results[self._asset_max_size_map[name]][0]
 
-                if asset in self.energy_system_components.get("heat_buffer", []):
+                if asset.name in self.energy_system_components.get("ates", []):
+                    asset.maxChargeRate = max(results[f"{name}.Heat_flow"])
+                    asset.maxDischargeRate = min(results[f"{name}.Heat_flow"])
```

Comment:
The max _size variable of the ATES could be used instaead of the max of the result of the heat_flow.

Reply:
Code has been updated